src/main/java/team/creative/creativecore/common/util/math/geo/VectorFan.java

```diff
@@ -955,31 +946,27 @@ else if (allValue)
         }
         
         if (left.size() >= 3 && done != null)
-            done.add(new VectorFan(left.toArray(new Vec3f[left.size()])));
+            done.add(new VectorFan(left.toArray(new Vec3f[0])));
```

Comment:
new Vec3f[0] will result in another array being created to match the size of the list.

Reply:
That's on purpose, the new array instance is not used and only required to cast array to the proper Class. Make the array with a pre established size wastes memory.

Reply:
ignore my stupidity, i will rever this damn change. that makes performance worse :P